File: examples/twosum.jl
function two_sum(nums::Vector{Int}, target::Int)
    num_to_index = Dict{Int, Int}()
    
    for (i, num) in enumerate(nums)
        complement = target - num
        if haskey(num_to_index, complement)
            return (num_to_index[complement], i)
        end
        num_to_index[num] = i
    end
    
    return nothing  # If no solution found
end

File: examples/twosum.rs
use std::collections::HashMap;

fn two_sum(nums: Vec<i32>, target: i32) -> Option<(usize, usize)> {
    let mut map = HashMap::new();
    
    for (i, num) in nums.iter().enumerate() {
        let complement = target - num;
        if let Some(&index) = map.get(&complement) {
            return Some((index, i));
        }
        map.insert(num, i);
    }
    
    None
}

fn main() {
    let nums = vec![2, 7, 11, 15];
    let target = 9;

    match two_sum(nums, target) {
        Some((i, j)) => println!("Indices: {} and {}", i, j),
        None => println!("No solution found."),
    }
}
